### PR TITLE
#6 Setup CD

### PR DESCRIPTION
**Pull request by michal-kucznerowicz**

### CODE REVIEW

File: .github/workflows/setup_build_environment.yml [hunk 1]
```diff
@@ -20,5 +20,9 @@ jobs:
         env:
           FIREBASE_GOOGLE_SERVICE: ${{ secrets.FIREBASE_GOOGLE_SERVICE }}
         run: echo $FIREBASE_GOOGLE_SERVICE > app/src/google-services.json
+      - name: Read service-credentials.json
+        env:
+          FIREBASE_SERVICE_CREDENTIALS: ${{ secrets.FIREBASE_SERVICE_CREDENTIALS }}
+        run: echo FIREBASE_SERVICE_CREDENTIALS > app/src/service-credentials.json
```
Comment (MarekMacko) [inline .github/workflows/setup_build_environment.yml]: Missing $

**Review by MarekMacko — CHANGES_REQUESTED** [review]

File: .github/workflows/setup_build_environment.yml [hunk 2]
```diff
@@ -20,5 +20,9 @@ jobs:
         env:
           FIREBASE_GOOGLE_SERVICE: ${{ secrets.FIREBASE_GOOGLE_SERVICE }}
         run: echo $FIREBASE_GOOGLE_SERVICE > app/src/google-services.json
+      - name: Read service-credentials.json
```
Comment (MarekMacko) [inline .github/workflows/setup_build_environment.yml]: It's rather `Save` than `Read`

File: .github/workflows/setup_build_environment.yml (hunk 1 above)
Comment (michal-kucznerowicz) [reply .github/workflows/setup_build_environment.yml]: Done

File: .github/workflows/setup_build_environment.yml (hunk 2 above)
Comment (michal-kucznerowicz) [reply .github/workflows/setup_build_environment.yml]: Done

File: .github/workflows/default_branch_verification.yml [hunk 3]
```diff
@@ -11,3 +11,9 @@ jobs:
     with:
       command: ./gradlew build
     secrets: inherit
+
+  app-distribution:
```
Comment (MarekMacko) [inline .github/workflows/default_branch_verification.yml:15]: minor: would be nice to have more explanatory name, e.g. `upload-debug-apk` or something similar, to clearly point what action will be executed

**Review by MarekMacko — APPROVED** [review]

File: .github/workflows/setup_build_environment.yml [hunk 4]
```diff
@@ -16,9 +16,13 @@ jobs:
...
         uses: android-actions/setup-android@v2.0.10
-      - name: Read google-service.json
+      - name: Save google-service.json
         env:
           FIREBASE_GOOGLE_SERVICE: ${{ secrets.FIREBASE_GOOGLE_SERVICE }}
         run: echo $FIREBASE_GOOGLE_SERVICE > app/src/google-services.json
+      - name: Save service-credentials.json
+        env:
+          FIREBASE_SERVICE_CREDENTIALS: ${{ secrets.FIREBASE_SERVICE_CREDENTIALS }}
+        run: echo $FIREBASE_SERVICE_CREDENTIALS > app/src/service-credentials.json
```
Comment (MarekMacko) [inline .github/workflows/setup_build_environment.yml:26]: I think we should rebuild this "basic" workflow, it's a nice example how basic script/class/etc. could grow too much.  `setup_build_environment` is also used for executing ktlint and tests, where it's totally redundant. Candidate for a separate issue.

Comment (michal-kucznerowicz) [reply .github/workflows/setup_build_environment.yml:26]: I fully agree. I've created an issue for that: https://github.com/intive/Picover/issues/208

File: .github/workflows/default_branch_verification.yml (hunk 3 above)
Comment (michal-kucznerowicz) [reply .github/workflows/default_branch_verification.yml:15]: Applied in PR https://github.com/intive/Picover/pull/202

**Review by marcinWisnia — APPROVED** [review]

**Pull request merged**